src/graph/optimizer/rule/IndexScanRule.cpp

```diff
@@ -199,46 +199,101 @@ Status IndexScanRule::appendIQCtx(const IndexItem& index, IndexQueryCtx& iqctx)
     }                                                                          \
   } while (0)
 
+inline bool verifyType(const Value& val) {
+  switch (val.type()) {
+    case Value::Type::__EMPTY__:
+    case Value::Type::NULLVALUE:
+    case Value::Type::VERTEX:
+    case Value::Type::EDGE:
+    case Value::Type::LIST:
+    case Value::Type::SET:
+    case Value::Type::MAP:
+    case Value::Type::DATASET:
+    case Value::Type::GEOGRAPHY:  // TODO(jie)
+    case Value::Type::PATH: {
+      DLOG(FATAL) << "Not supported value type " << val.type() << "for index.";
+      return false;
+    } break;
+    default: {
+      return true;
+    }
+  }
+}
 Status IndexScanRule::appendColHint(std::vector<IndexColumnHint>& hints,
                                     const FilterItems& items,
                                     const meta::cpp2::ColumnDef& col) const {
+  // CHECK(false);
   IndexColumnHint hint;
-  Value begin, end;
+  std::pair<Value, bool> begin, end;
   bool isRangeScan = true;
   for (const auto& item : items.items) {
+    if (!verifyType(item.value_)) {
+      return Status::SemanticError(
+          fmt::format("Not supported value type {} for index.", item.value_.type()));
+    }
     if (item.relOP_ == Expression::Kind::kRelEQ) {
       // check the items, don't allow where c1 == 1 and c1 == 2 and c1 > 3....
       // If EQ item appears, only one element is allowed
       if (items.items.size() > 1) {
         return Status::SemanticError();
       }
       isRangeScan = false;
-      begin = OptimizerUtils::normalizeValue(col, item.value_);
+      begin = {item.value_, true};
       break;
     }
     // because only type for bool is true/false, which can not satisify [start,
     // end)
     if (col.get_type().get_type() == nebula::cpp2::PropertyType::BOOL) {
       return Status::SemanticError("Range scan for bool type is illegal");
     }
-    NG_RETURN_IF_ERROR(OptimizerUtils::boundValue(item.relOP_, item.value_, col, begin, end));
```

Comment:
Delete all boundValue related functions.

Reply:
There are some other functions that still use them, such as `mergeRangeColumnHints`. The logic of these functions is very complicated, so I didn't modify them.

Reply:
If this function is called, isn't the value it changes illegal？
If we add an interval open and close flag, we shouldn't do boundValue anymore.

Reply:
OK, I have deleted `boundValue` just now, but the logic of `selectIndex` is very complicated, and there are a lot of code repetitions. I am more inclined to reorganize this part of the logic when you reconstruct the logic of selecting indexes.

Reply:
Ok.